src/components/ActionMenu/components/Actions/Actions.tsx

```diff
@@ -35,32 +56,92 @@ export function Actions({actions = [], groups = []}: Props) {
     () => setActiveMenuGroup(undefined),
     [],
   );
-  const menuActions = [...actions, ...groups];
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
It would be better to avoid disabling this if we can. I imagine this is complaining because if the `setAvailableWidth` changed it would make sure this callback gets updated to include the new reference.

Reply:
This is because we're passing in debounce and useCallback can't determine the dependencies within. I don't think we can get around this or at least we weren't able to here as well: https://github.com/Shopify/polaris-react/blob/master/src/components/MediaQueryProvider/MediaQueryProvider.tsx#L19-L31

Reply:
Would useCallBack work the same if we re-ordered them like

```tsx
  const handleResize = debounce(
    useCallback(() => {
      if (!newDesignLanguage || !actionsLayoutRef.current) return;
      setAvailableWidth(actionsLayoutRef.current.offsetWidth);
    }, [newDesignLanguage]),
    20,
    {leading: false, trailing: true, maxWait: 40},
  );
```

Reply:
Ah right, I remember this from our pairing now. I think it isn't quite the same with that because handleResize would still setup a new debounce on each render. Probably fine to ignore and keep your disabling 👍

Reply:
Hooks can only be inside hooks, or at the root of a component ;)

You can replace `useCallback` with `useMemo(() => debounce(...` and it should be able to determine the dependencies I believe.

Reply:
I see the same warning with use memo

I could just pass in a function...

```tsx
  const debouncedSetWidth = debounce(
    function () {
      if (!newDesignLanguage || !actionsLayoutRef.current) return;
      setAvailableWidth(actionsLayoutRef.current.offsetWidth);
    },
    20,
    {leading: false, trailing: true, maxWait: 40},
  );

  const handleResize = useCallback(debouncedSetWidth, [debouncedSetWidth]);
```

Reply:
Sorry I'm a bit late, but I wanted to clarify something :)
You still see a warning, but it's a different one.

Here's the warning you see with `useCallback` and debounce:

> React Hook useCallback received a function whose dependencies are unknown. Pass an inline function instead.eslintreact-hooks/exhaustive-deps

Here's the kind of warning you'll see with `useMemo` (if some deps are missing):

> React Hook useMemo has a missing dependency: 'newDesignLanguage'. Either include it or remove the dependency array.eslintreact-hooks/exhaustive-deps

`newDesignLanguage` is not really expected to change in the lifetime of the app, so it's "ok", but otherwise this wouldn't do what we want it to do imo ;)

![image](https://user-images.githubusercontent.com/1972567/94810072-89c6a300-03c1-11eb-9596-23fea7d05ba3.png)